packages/twenty-server/src/engine/api/graphql/workspace-query-runner/workspace-query-runner.service.ts

```diff
@@ -305,11 +298,61 @@ export class WorkspaceQueryRunnerService {
     return parsedResults;
   }
 
+  async upsertMany<Record extends IRecord = IRecord>(
+    args: CreateManyResolverArgs<Record>,
+    options: WorkspaceQueryRunnerOptions,
+  ): Promise<Record[] | undefined> {
+    const results = await Promise.all(
+      args.data.map(async (payload) => {
+        const existingRecord = payload.id
+          ? await this.findOne(
+              { filter: { id: { eq: payload.id } } } as FindOneResolverArgs,
+              options,
+            )
+          : null;
+
+        if (existingRecord) {
+          const result = await this.updateOne(
```

Comment:
Same, I feel like we could optimise using batch queries 🤔 Although you are using Promise.all 🙂

Reply:
Actually, updateMany will be limited in terms of number of affected rows due to MUTATION_MAXIMUM_RECORD_AFFECTED so you would have to take this into account and do it in batches